docs/lang/articles/basic/type.md

```diff
@@ -1,217 +1,188 @@
+### Default primitive types for integers and floating-point numbers
 
-(:large_orange_diamond: Requiring extensions of the backend)
+- The default integer type in Taichi is `ti.i32`.
+- The default floating-point type in Taichi is `ti.f32`.
 
-### Default types for integers and floating-point numbers
+#### Customize the default type
 
-An integer literal, e.g., `42`, has default type `ti.i32`, while a floating-point literal,
-e.g., `3.14`, has default type `ti.f32`. This behavior can be changed by explicitly specifying
-default types when initializing Taichi:
+You can change the default primitive types by explicitly specifying the default types when initializing Taichi:
 
 ```python
-ti.init(default_ip=ti.i64)  # set default integer type to ti.i64
-ti.init(default_fp=ti.f64)  # set default floating-point type to ti.f64
+ti.init(default_ip=ti.i64)  # Set the default integer type to ti.i64
+ti.init(default_fp=ti.f64)  # Set the default floating-point type to ti.f64
 ```
 
-In addition, you can use `int` as an alias for the default integer type, and `float` as an alias
-for the default floating-point type:
+#### Set alias
+
+Taichi supports using `int` as an alias for the default integer type and `float` as an alias for the default floating-point type. In the following example, you change the default primitive types to `i64` and `f64` when initializing Taichi, then you can use `int` to represent `i64` and `float` to represent `f64`.
 
 ```python
-ti.init(default_ip=ti.i64, default_fp=ti.f32)
+ti.init(default_ip=ti.i64, default_fp=ti.f64)
 
 x = ti.field(float, 5)
 y = ti.field(int, 5)
 # is equivalent to:
-x = ti.field(ti.f32, 5)
+x = ti.field(ti.f64, 5)
 y = ti.field(ti.i64, 5)
 
 def func(a: float) -> int:
     ...
 # is equivalent to:
-def func(a: ti.f32) -> ti.i64:
+def func(a: ti.f64) -> ti.i64:
     ...
 ```
 
 ### Explicit type casting
 
-Just like programming in other languages, you may encounter situations where you have a certain
-type of data, but it is not feasible for the assignment or calculation you want to perform. In this
-case, you can do *explicit type casting*. There are two kinds of explicit type casting in Taichi,
-namely *normal casting* and *bit casting*.
-
-:::caution
-In Taichi-scope, the type of a variable is **static** and **determined on its initialization**.
-That is, you can never change the type of a variable. The compiler relies on this compile-time
-information to check the validity of expressions in Taichi programs.
-:::
+In the Taichi scope, the type of a variable is *static* and *completely determined when it is initialized*. Taichi's compiler does type check at compile time, so you *cannot* change a variable's type once it is initialized. Still, from time to time, you may run into a situation where you have a certain type of data but it is not feasible for an assignment or calculation. Then, you need explicit type casting:
```

Comment:
Note that statically typed is the idiomatic wording to describe a programming language.

```suggestion
In the Taichi scope, a variable is *statically typed* upon initialization. Taichi's compiler does type check at compile time, so you *cannot* change a variable's type once it is initialized. Still, from time to time, you may run into a situation where you have a certain type of data but it is not feasible for an assignment or calculation. Then, you need explicit type casting:
```